### PR TITLE
Transition to pure pyproject.toml for project metadata 

### PR DESCRIPTION
**Pull request by gsheni**

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -4,6 +4,7 @@ Release Notes
 Future Release
 ==============
     * Enhancements
+        * Transition to pure pyproject.toml for project metadata (:pr:`351`)
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:7]: Enhancement or change?

File: .github/workflows/install_test.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
-        python_version: ["3.8", "3.9", "3.10"]
+        python_version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (thehomebrewnerd) [inline .github/workflows/install_test.yml:16]: If we are going to support 3.11, shouldn't we also run all our unit tests with 3.11, not just the install test?

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,150 @@
...
+    "nbconvert == 6.4.5",
+    "ipython == 7.31.1",
+    "pygments == 2.10.0",
+    "jupyter == 1.0.0",
+    "pandoc == 1.1.0",
+    "ipykernel == 6.4.2",
+    "scikit-learn >= 0.20.0, !=0.22",
+]
+complete = [
+    "featuretools[updater]",
```
Comment (thehomebrewnerd) [inline pyproject.toml:80]: featuretools -> compose

File: pyproject.toml [hunk 4]
```diff
@@ -1,6 +1,150 @@
...
+    "woodwork >= 0.11.0",
+    "pyarrow >= 3.0.0",
+]
+updater = [
+    "alteryx-open-src-update-checker >= 2.1.0"
+]
+docs = [
+    "evalml >= 0.45.0"
+]
+dev = [
```
Comment (thehomebrewnerd) [inline pyproject.toml:60]: Previously the dev requirements also included the test requirements. Should we also add `compose[test]` in this section?

**Review by ozzieD — APPROVED** [review]

**Review by thehomebrewnerd — APPROVED** [review]

**Pull request merged**